dracut/30afterburn/afterburn-hostname.service

```diff
@@ -16,4 +16,6 @@ OnFailureJobMode=isolate
 
 [Service]
 ExecStart=/usr/bin/afterburn --cmdline --hostname=/sysroot/etc/hostname
+ExecStart=/bin/sh -c 'mkdir -p /sysroot/run/tmpfiles.d'
+ExecStart=/bin/sh -c 'echo "Z /etc/hostname - - -" > /sysroot/run/tmpfiles.d/hostname-relabel.conf'
```

Comment:
You'll likely also need to order whatever problematic services to *after* `systemd-tmpfiles-setup.service`. See e.g. https://github.com/coreos/fedora-coreos-config/blob/1842e9e309c2dfc89863705917c19f8eb85be245/overlay/usr/lib/dracut/modules.d/40coreos-var/coreos-populate-var.sh#L39. Depending on the service, that might be tricky to do.

Reply:
Shouldn't be an issue since AFAIK only NM should be using this

Reply:
Hmm, does NM have an explicit ordering to `systemd-tmpfiles-setup.service`?

Reply:
Well, none of the NM units use `DefaultDependencies=no`, so it should already be ordered after `sysinit.target` (which pulls in `systemd-tmpfiles-setup.service`), so yeah, I think we're fine.